doc/releases/release-notes-3.5.rst

```diff
@@ -177,6 +177,23 @@ Drivers and Sensors
 
 * IEEE 802.15.4
 
+  * A new mandatory method attr_get() was introduced into ieee802154_radio_api.
+    Drivers need to implement at least
+    IEEE802154_ATTR_PHY_SUPPORTED_CHANNEL_PAGES and
+    IEEE802154_ATTR_PHY_SUPPORTED_CHANNEL_RANGES.
+  * The hardware capabilities IEEE802154_HW_2_4_GHZ and IEEE802154_HW_SUB_GHZ
+    were removed as they were not aligned with the standard and some already
+    existing drivers couldn't properly express their channel page and channel
+    range (notably SUN FSK and HRP UWB drivers). The capabilities were replaced
+    by the standard conforming new driver attribute
+    IEEE802154_ATTR_PHY_SUPPORTED_CHANNEL_PAGES that fits all in-tree drivers.
+  * The method get_subg_channel_count() was removed from ieee802154_radio_api.
+    This method could not properly express the channel range of existing drivers
+    (notably SUN FSK drivers that implement channel pages > 0 and may not have
+    zero-based channel ranges or UWB drivers that could not be represented at
+    all). The method was replaced by the new driver attribute
+    IEEE802154_ATTR_PHY_SUPPORTED_CHANNEL_RANGES that fits all in-tree drivers.
+
```

Comment:
It looks to me like a stable API Change (IEEE 802.15.4 driver interface API), should not part of it be added to  “API Changes” section of the release notes? To be fear there is not “API Changes” section in `doc/releases/release-notes-3.4.rst`.

I also could not find an RFC mail sent to the devel list about this, is there a new undocumented process or is it not needed here?

Reply:
That's a good point. However, I can't find the ieee802154 marked as stable here:
https://docs.zephyrproject.org/latest/develop/api/overview.html

>should not part of it be added to “API Changes” section of the release notes?

Stable API changes now belong in the migration guide, not in the release notes (hence why no section).

@fgrandel should we consider this API as unstable for now?

Reply:
@carlescufi Yes, one of the PRs blocked by this PR does just that. :-)

See #62503

The documentation has been less than satisfactory so far. And that PR fixes a lot of it.

Reply:
> This is part of networking APIs, stable since 1.0

I personally don't agree. The subsystem exposes three independent APIs: L2, driver and net_mgmt (each with their corresponding configuration approach in Kconfig, DT and runtime). As these three APIs have very different rythms of change and clients/exposure in applications I propose we treat those as separate.

@rlubos, @jukkar  and I considered the existing user base of those APIs small and the changes required for its transformation many so that we agreed to treat it de facto as unstable over the last three releases (which I'm proposing to formalize now).

This strategy seems to have worked out well. AFAICS we didn't get a single complaint and a lot of positive feedback for that approach from the community.

Reply:
> This is part of networking APIs, stable since 1.0 https://docs.zephyrproject.org/latest/develop/api/overview.html#api-overview [Networking APIs](https://docs.zephyrproject.org/latest/connectivity/networking/api/index.html#networking-api)

Unfortunately I don't think it's all that clear, and we don't really have a single source of truth for what's the status of each API within the Networking area.
For example that page you're linking also has CoAP, or MQTT, that are clearly labelled unstable, and many other entries that just don't happen to have ever been properly qualified in the documentation (ex. TFTP seems to be experimental if one looks at the Kconfig, but what should one make of it being listed on that page with no clear warning it's experimental?)

Reply:
I agree with the statements from @fgrandel and @kartben, but I will ask @rlubos and @jukkar to weigh in.

Reply:
Hi @carlescufi ,

Just a remind that this IEEE 802.15.4 don't follow 100% the standard and many issues have been addressed recently. I would say the API is far to be stable and there are many changes in the backlog (which is not visible yet). My attempt to give the idea about a central point that have the features already in/planned goes in the direction to create some visibility about work already done and planned. As an example, the #49775 is already an example about things that we have been discussing over time and that already require possible changes/extensions in the API.

Reply:
okay, it convinces me that it is okay to cherry-pick Radio API to be unstable.
But then why cannot [this change](https://github.com/zephyrproject-rtos/zephyr/pull/62503/files#diff-925c8ea16c15775440380c5e134bba4f8ba66aed774012b72b149e1b2210b884) be a commit before breaking changes, for the avoidance of any doubts?

Reply:
I agree, the list of different components constituting "Networking API" is just too broad to have it covered by a single entry.